pandas/__init__.py

```diff
@@ -186,10 +183,32 @@
 
 
 # GH 27101
+__deprecated_num_index_names = ["Float64Index", "Int64Index", "UInt64Index"]
+
+
+def __dir__():
```

Comment:
why is this needed?

Reply:
This is so Int64Index etc. show up in dir calls. Int64Index etc. have been quite prominent parts of the public API, so I think it's better that they still show up for the duration of Pandas 1.x. This should be removed in Pandas 2.0.

Reply:
ok can you add a comment to this effect